### ISSUE TITLE
Add ability to pass flags to Node instance

### ISSUE THREAD

**Issue by crutchcorn** [nodejs/single-executable#100]
I am trying to write a program that accesses the `internal` package from Node.js. To do this in most apps I would typically pass:

[--expose-internals](https://github.com/nodejs/node/blob/v20.2.0/src/node_options.cc#L448)

To handle this edgecase.

However, it seems that doing this after-the-executable generation does not appear to do anything.

```shell
.\its-node.exe --expose-internals
node:internal/util/embedding:48
    throw new ERR_UNKNOWN_BUILTIN_MODULE(id);
    ^

Error [ERR_UNKNOWN_BUILTIN_MODULE]: No such built-in module: internal/fs/utils
    at embedderRequire (node:internal/util/embedding:48:11)
    at ./out/bundle.js:3197:5
    at embedderRunCjs (node:internal/util/embedding:37:10) {
  code: 'ERR_UNKNOWN_BUILTIN_MODULE'
}

Node.js v22.2.0
```

Moreover, it would be nice if my users didn't have to pass a flag of any kind, as the option was enabled from the executable to begin with.

Thanks for your efforts on this and taking the time to read through the request 😊

**Comment by crutchcorn** [comment]
Right, but I'd love for the `NODE_OPTIONS` to be embedded _into_ `./server`

**Comment by jasonolmstead33** [comment]
> Right, but I'd love for the `NODE_OPTIONS` to be embedded _into_ `./server`

ah yes, I agree. make it part of the blob when building the SEA or something

**Comment by joyeecheung** [comment]
I wrote some thoughts about how to support this in https://github.com/nodejs/node/issues/51688#issuecomment-3144431518
